TFUtil.py

```diff
@@ -8651,16 +8651,26 @@ def get_positional_encoding(num_channels, length=None, position=None, min_timesc
     position = tf.range(length)
   else:
     assert length is None
-    position.set_shape((None,))
+    if len(position.shape) == 1:
+      position.set_shape((None,))
   position = tf.to_float(position)
   num_timescales = num_channels // 2
   log_timescale_increment = (
     math.log(float(max_timescale) / float(min_timescale)) / (float(num_timescales - 1)))
   inv_timescales = min_timescale * tf.exp(
     tf.to_float(tf.range(num_timescales)) * -log_timescale_increment)
-  scaled_time = tf.expand_dims(position, 1) * tf.expand_dims(inv_timescales, 0)
-  signal = tf.concat([tf.sin(scaled_time), tf.cos(scaled_time)], axis=1)
-  signal = tf.pad(signal, [[0, 0], [0, num_channels % 2]])  # (length, channels)
+  if len(position.shape) == 1:
+    scale = tf.expand_dims(inv_timescales, 0)
+  else:
+    assert len(position.shape) == 2
+    scale = tf.expand_dims(tf.expand_dims(inv_timescales, 0), 0)
```

Comment:
Use `expand_multiple_dims` or `tf.reshape`. This is ugly otherwise.
Also, comment about the shape. (`# (1, 1, D//2)`)
Also, why don't you simplify the code, and cover any number of axes in `position`?
E.g. like this:
```
scale = tf.reshape(inv_timescales, [1] * len(position.shape) + [num_timescales])
```

Reply:
Done.